### PR TITLE
Add "Apply" button to the Options dialog

### PR DESCRIPTION
**Pull request by mkow**
I also refactored a bit the way we reload settings.

### CODE REVIEW

**Comment by mkow** [comment]
Fixes #389.

**Comment by koriakin** [comment]
Are you sure using an interface is the right idea? It seems to be a perfect opportunity to use signals:

- have a top-level settings object with a "settings changed" signal
- have the options dialog emit that signal
- in the constructor of every settings-using widget, connect the signal to an internal slot

**Comment by mkow** [comment]
I'm afraid that this solution only sounds good, but in practice will be much less nice.
Reasons:
* Non-POD globals in C++ are dangerous in general (e.g. GCS forbids them: https://google.github.io/styleguide/cppguide.html#Static_and_Global_Variables)
* Global QObjects are even more dangerous (e.g. because they outlive the main QApplication instance). Some things can probably break during application exit.

I also don't like the idea of scanning all widgets and calling a method on them, but this still seems to be the safest solution to me. So, in general, on the high-level I prefer your solution because it seems cleaner, but it seems that C++ will make it hard to implement safely.

Other solution would be to define a signal in `MainWindowWithDetachableDockWidgets` and emit it for every main window, but then we assume that all of our main windows derive from `MainWindowWithDetachableDockWidgets`, which probably won't hold in the future, and causes a lot of troubles when moving widgets between main windows.

**Comment by malpunek** [comment]
There might be a solution for static non-PODs here - https://isocpp.org/wiki/faq/ctors#static-init-order-on-first-use

Apart from that I don't really see why `options` should be static. Why couldn't we subclass `QApplication` to include extra members like `private Options options; public signal optionsChanged();`? Or maybe we can treet `options` as a special kind of applications style (QStyle) and then simply use `QApplication::setStyle(options)`?

File: include/ui/velesapplication.h [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+namespace veles {
+namespace ui {
+
+// A modified QApplication which stores an additional global state.
+class VelesApplication : public QApplication {
+  Q_OBJECT
+
+public:
+  using QApplication::QApplication;
```
Comment (malpunek) [inline include/ui/velesapplication.h]: Is this necessary?

File: src/ui/optionsdialog.cc [hunk 2]
```diff
@@ -79,20 +82,17 @@ void OptionsDialog::accept() {
       color_3d_begin_button_->getColor());
```
Comment (malpunek) [inline src/ui/optionsdialog.cc]: I know this is not your code, but shouldn't these be slots of HexEdit & Visualization connected to settingsChanged signal?

File: include/ui/velesapplication.h (hunk 1 above)
Comment (mkow) [reply include/ui/velesapplication.h]: Yes. This line inherits a constructor which is used in main().

File: src/ui/optionsdialog.cc (hunk 2 above)
Comment (mkow) [reply src/ui/optionsdialog.cc]: No, as explained in private.

**Comment by malpunek** [comment]
Czy to nie fixuje czasem #390 ?

**Comment by mkow** [comment]
Right :)

Fixes #390.

**Pull request merged**